cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -2876,6 +2881,11 @@
 (rule (x64_mfence)
       (SideEffectNoResult.Inst (MInst.Fence (FenceKind.MFence))))
 
+(decl x64_cmpxchg (Type Gpr Gpr SyntheticAmode) Gpr)
+(rule (x64_cmpxchg ty expected replacement addr)
+      (let ((_ Unit (emit (MInst.LockCmpxchg ty replacement expected addr (rax)))))
+        (rax)))
```

Comment:
It feels like this instruction should be proclaiming the fact that it has a side effect but `SideEffectNoResult` doesn't really fit here.

Reply:
I think it's probably fine -- the main point of `SideEffect` was to provide a return value for a thing that nominally doesn't have one, like a store; but here we have a natural value to return. Perhaps the enum is a little misnamed but I can't think of anything better at the moment...